tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -95,7 +102,26 @@ def parse_datafile(datafile, dfs):
         bulk_create_errors(unsaved_parser_errors, 1, flush=True)
         return errors
 
-    line_errors = parse_datafile_lines(datafile, dfs, program_type, section, is_encrypted, case_consistency_validator)
+    # Last resort to catch any un-caught exceptions during parsing and handle state appropriately
+    line_errors = {}
+    try:
+        line_errors = parse_datafile_lines(datafile, dfs, program_type, section,
+                                           is_encrypted, case_consistency_validator)
+    except Exception as e:
+        # TODO: is this the best way to force the datafile to be rejected while also having an error report?
+        generate_error = util.make_generate_parser_error(datafile, None)
+        error = generate_error(schema=None,
+                               error_category=ParserErrorCategoryChoices.PRE_CHECK,
+                               error_message="Uncaught parsing exception, rejecting file.",
+                               record=None,
+                               field=None
+                               )
+        error.save()
```

Comment:
I would love some feedback on a better way to force the file into a rejected state along with generating some kind of error report if no errors have been generated yet. I don't love using `ParserErrorCategoryChoices.PRE_CHECK` to identify this error.

Reply:
I do agree with you that RPE_CHECK is misleading here. 

My suggestion is to add another category to ParserErrorCategoryChoices

Also, another question I have is where are we logging {e}?

Reply:
My thinking was to add maybe even a few more options to the ParserError choices. Ill bring it up in OH and we can refine.

The exception gets logged to the console and DAC on line `121`